manage_breast_screening/auth/views.py

```diff
@@ -78,3 +86,65 @@ def jwks(request):
         return JsonResponse({"keys": [jwk_dict]})
     except Exception:
         return JsonResponse({"keys": []})
+
+
+@require_http_methods(["POST"])
+@csrf_exempt
+@login_not_required
+def cis2_back_channel_logout(request):
+    """Handle CIS2 back-channel logout notifications.
+
+    This endpoint receives logout tokens from CIS2 when a user's session is terminated.
+    See: https://openid.net/specs/openid-connect-backchannel-1_0.html
+    See: https://digital.nhs.uk/services/care-identity-service/applications-and-services/cis2-authentication/guidance-for-developers/detailed-guidance/session-management#back-channel-logout
+    """
+    logout_token = request.POST.get("logout_token")
+    if not logout_token:
+        return HttpResponseBadRequest("Missing logout_token")
+
+    # Get the CIS2 client and prepare key loader for token verification
+    client = get_cis2_client()
+    metadata = client.load_server_metadata()
+    key_loader = client.create_load_key()
+    try:
+        verification_rules = {
+            "iss": {"values": [metadata["issuer"]], "essential": True},
+            "aud": {"values": [client.client_id], "essential": True},
+            "exp": {"essential": True},
+            "iat": {"essential": True},
+            "events": {
+                "essential": True,
+                "validate": lambda claim, value: isinstance(value, dict)
+                and "http://schemas.openid.net/event/backchannel-logout" in value,
+            },
+            "nonce": {"validate": lambda claim, value: value is None},
+        }
+        claims = jwt.decode(
+            logout_token,
+            key=key_loader,
+            claims_options=verification_rules,
+        )
+        claims.validate(leeway=60)
+    except (JoseError, InvalidClaimError, ExpiredTokenError, InvalidTokenError) as e:
+        logging.error(f"Invalid logout token: {str(e)}")
```

Comment:
Two things on logging:

- there is a general convention to create a logger per module which is preferable to using the global functions - this creates a named logger rather than sending everything to the root logger, which is more useful for configuration purposes

  Example from the docs:

  ```python
  # mylib.py
  import logging
  logger = logging.getLogger(__name__)

  def do_something():
      logger.info('Doing something')
  ```


- There is an [`exception` method](https://docs.python.org/3/library/logging.html#logging.Logger.exception) you can use instead of `error`, which captures the exception object with the full stack trace. IIRC the default formatter the output is the same, but it just gives the flexibility to do something with that later.